bundles/mapping/drawtools/plugin/DrawPlugin.ol3.js

```diff
@@ -166,14 +173,17 @@ Oskari.clazz.define(
          * @param {String} id
          * @param {boolean} clearCurrent: if true, all selection will be removed from the map
          */
-        stopDrawing : function(id, clearCurrent) {
+        stopDrawing : function(id, clearCurrent, supressEvent) {
             var me = this;
+            if( typeof supressEvent === undefined ) {
+                supressEvent = false;
+            }
             var options = {
                 clearCurrent: clearCurrent,
                 isFinished: true
             };
             if(me._functionalityIds[id]) {
-                me.sendDrawingEvent(id, options);
+                supressEvent === true ? me.clearDrawing(id) : me.sendDrawingEvent(id, options);

```

Comment:
So if event isn't sent the clearDrawing() isn't called? Looks like a bug. See this for previous example of using suppressEvent: https://github.com/oskariorg/oskari-frontend/blob/master/bundles/mapping/mapmodule/AbstractMapModule.js#L693-L703

Reply:
sendDrawingEvent clears the drawing also if it's specified in the options: https://github.com/nxholm/oskari-frontend/blob/41d293710a66673f75b736569436434832bb3945/bundles/mapping/drawtools/plugin/DrawPlugin.ol3.js#L244